controller/lib/core/control.cpp

```diff
@@ -57,89 +54,35 @@ void pid_init() {
 
 void pid_execute(const VentParams &params, SensorReadings *readings) {
   static uint16_t cyclecounter = 0;
-  static enum pid_fsm_state state = pid_fsm_state::reset;
-
-  // The state machine drives PID to blow a trapezoid.  We start at the PEEP
-  // pressure, then gradually ramp the setpoint up to PIP.  Then we dwell there
-  // for a while, ramp down to PEEP, and dwell there for a bit.
-  //
-  // TODO(jlebar): Per Edwin, we should simplify this by removing the ramp-up
-  // and ramp-down states.  Instead, just modify the setpoint and let PID ramp
-  // us up/down.  The timescales for the ramp-up and ramp-down are short, and
-  // PID only moves us so fast anyway.
-  switch (state) {
-  case pid_fsm_state::reset: // reset
-    cyclecounter = 0;
-    Setpoint = PEEP;
-    state = pid_fsm_state::inspire; // update state
-    break;
-
-  case pid_fsm_state::inspire: // Inspire
-    cyclecounter++;
-    // set command
-    Setpoint += INSPIRE_RATE;
-    if (Setpoint > PIP) {
-      Setpoint = PIP;
-    }
-    // update state
-    if (cyclecounter > INSPIRE_TIME) {
-      cyclecounter = 0;
-      state = pid_fsm_state::plateau;
-    }
-    break;
+  static enum pid_fsm_state state = pid_fsm_state::EXPIRE;
 
-  case pid_fsm_state::plateau: // Inspiratory plateau
+  // The state machine drives PID to blow a square wave.  The PID setpoint
+  // moves from PEEP pressure (EXPIRE state) to PIP pressure (INSPIRE state)
+  // and back to PEEP.
+  switch (state) {
+  case pid_fsm_state::INSPIRE:
     cyclecounter++;
```

Comment:
Are we assuming here that all cycles take the same amount of time? Do we know how long, and is that device-independent?

Also, cycles probably don't actually take the same amount of time unless we're scheduling this code to run on a timer - are we?